project_setup/scripts/iam-to-github

```diff
@@ -0,0 +1,247 @@
+    args: Dict[str, str] = docopt.docopt(__doc__, version="1.0.0")
+
+    # Validate and convert arguments as needed
+    schema: Schema = Schema(
+        {
+            "<github-personal-access-token>": And(
+                str,
+                lambda n: len(n) == 40,
+                error="<github-token> must be a 40 character personal access token.",
+            ),
+            "--log-level": And(
+                str,
+                Use(str.lower),
+                lambda n: n in ("debug", "info", "warning", "error", "critical"),
+                error="Possible values for --log-level are "
+                + "debug, info, warning, error, and critical.",
+            ),
+            "--repo": Or(
+                None,
+                And(
+                    str,
+                    lambda n: "/" in n,
+                    error='Repository names must contain a "/"',
+                ),
+            ),
+            "--suffix": Or(And(None, Use(lambda n: "")), And(str, Use(str.upper))),
+            str: object,  # Don't care about other keys, if any
+        }
+    )
+
+    try:
+        validated_args: Dict[str, Any] = schema.validate(args)
+    except SchemaError as err:
+        # Exit because one or more of the arguments were invalid
+        print(err, file=sys.stderr)
+        return 1
+
+    # Assign validated arguments to variables
+    github_token: str = validated_args["<github-personal-access-token>"]
+    log_level: str = validated_args["--log-level"]
+    repo_name: str = validated_args["--repo"]
+    secret_suffix: str = validated_args["--suffix"]
+    user_filter: str = validated_args["--user"]
+
+    # Set up logging
+    logging.basicConfig(
+        format="%(asctime)-15s %(levelname)s %(message)s", level=log_level.upper()
+    )
+
+    # If the user does not provide a repo name we'll try to determine it from git
+    if not repo_name:
+        repo_name = get_repo_name()
+    logging.info(f"Using GitHub repository name: {repo_name}")
+
+    if user_filter:
+        logging.info(f"Filter to users containing: {user_filter}")
+
+    if secret_suffix:
+        if not secret_suffix.startswith("_"):
+            secret_suffix = f"_{secret_suffix}"
+        logging.info(f'Apending "{secret_suffix}" to secret names.')
+
+    aws_user: Optional[str] = None
+    aws_key_id: Optional[str] = None
+    aws_secret: Optional[str] = None
+    cred_list: List[Tuple[str, str, str]] = []
+
+    logging.info("Searching Terraform state for IAM credentials.")
+    for aws_user, aws_key_id, aws_secret in creds_from_terraform():
+        logging.info(f"Found credentials for user: {aws_user}")
+        if not user_filter or user_filter in aws_user:
+            logging.debug(f"User {aws_user} matches filter {user_filter}")
+            cred_list.append((aws_user, aws_key_id, aws_secret))
+
+    if len(cred_list) == 0:
+        logging.error("No credentials matched in terraform state.")
```

Comment:
Inconsistent capitalization of "terraform" vs. "Terraform" here.

Reply:
Fixed in: 381953d